samples/seat_massager/provider/src/provider_impl.rs

```diff
@@ -0,0 +1,200 @@
+}
+
+impl ProviderImpl {
+    fn set_message_airbags(
+        properties: Arc<Mutex<ProviderProperties>>,
+        value: &str,
+    ) -> Result<(), String> {
+        let message_airbags_property_json: serde_json::Value = serde_json::from_str(value).unwrap();
+        let message_airbags_json = message_airbags_property_json
+            .get(sdv::airbag_seat_massager::massage_airbags::NAME)
+            .unwrap();
+
+        let massage_airbags: sdv::airbag_seat_massager::massage_airbags::TYPE =
+            serde_json::from_value(message_airbags_json.clone()).unwrap();
+
+        info!("Setting message airbags to: {:?}", massage_airbags);
+
+        // This block controls the lifetime of the lock.
+        {
+            let mut lock: MutexGuard<ProviderProperties> = properties.lock();
+
+            lock.massage_airbags = massage_airbags;
+        }
+
+        Ok(())
+    }
+
+    fn get_message_airbags(properties: Arc<Mutex<ProviderProperties>>) -> Result<String, String> {
+        let mut property: Property = Property {
+            massage_airbags: Vec::new(),
+            metadata: Metadata {
+                model: sdv::airbag_seat_massager::massage_airbags::ID.to_string(),
+            },
+        };
+
+        // This block controls the lifetime of the lock.
+        {
+            let lock: MutexGuard<ProviderProperties> = properties.lock();
+
+            property.massage_airbags = lock.massage_airbags.clone();
+        }
+
+        let json_str = serde_json::to_string(&property).unwrap();
+
+        Ok(json_str)
+    }
+}
+
+#[tonic::async_trait]
+impl DigitalTwinProvider for ProviderImpl {
+    /// Subscribe implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Subscribe request.
+    async fn subscribe(
+        &self,
+        request: Request<SubscribeRequest>,
+    ) -> Result<Response<SubscribeResponse>, Status> {
+        warn!("Got subscribe request: {request:?}");
+
+        Err(Status::unimplemented("subscribe has not been implemented"))
+    }
+
+    /// Unsubscribe implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Unsubscribe request.
+    async fn unsubscribe(
+        &self,
+        request: Request<UnsubscribeRequest>,
+    ) -> Result<Response<UnsubscribeResponse>, Status> {
+        warn!("Got an unsubscribe request: {request:?}");
+
+        Err(Status::unimplemented("unsubscribe has not been implemented"))
+    }
+
+    /// Get implementation.
+    ///
+    /// # Arguments
+    /// * `request` - Get request.
+    async fn get(&self, request: Request<GetRequest>) -> Result<Response<GetResponse>, Status> {
+        let request_inner = request.into_inner();
+        let entity_id: String = request_inner.entity_id.clone();
+        let consumer_uri: String = request_inner.consumer_uri;
+
+        info!("Received a get request for entity id {entity_id}");
+
+        let properties: Arc<Mutex<ProviderProperties>> = self.properties.clone();
```

Comment:
the type annotation here is not necessary

Reply:
I prefer to make it explicit.  Leaving it as is.